diff --git a/include/core/SkMatrix.h b/include/core/SkMatrix.h
index 02c873623b..be12953daa 100644
--- a/include/core/SkMatrix.h
+++ b/include/core/SkMatrix.h
@@ -96,7 +96,7 @@ public:
         return m;
     }
     [[nodiscard]] static SkMatrix Translate(SkVector t) { return Translate(t.x(), t.y()); }
-    [[nodiscard]] static SkMatrix Translate(SkIVector t) { return Translate(t.x(), t.y()); }
+    [[nodiscard]] static SkMatrix Translate(SkIVector t) { return Translate((SkScalar) t.x(), (SkScalar) t.y()); }
 
     [[nodiscard]] static SkMatrix ScaleTranslate(float sx, float sy, float tx, float ty);
 
