diff --git a/include/codec/SkEncodedOrigin.h b/include/codec/SkEncodedOrigin.h
index 19d083672f..9f4923e147 100644
--- a/include/codec/SkEncodedOrigin.h
+++ b/include/codec/SkEncodedOrigin.h
@@ -29,7 +29,9 @@ enum SkEncodedOrigin {
  * that transforms the source rectangle with upper left corner at [0, 0] and origin to a correctly
  * oriented destination rectangle of [0, 0, w, h].
  */
-static inline SkMatrix SkEncodedOriginToMatrix(SkEncodedOrigin origin, int w, int h) {
+static inline SkMatrix SkEncodedOriginToMatrix(SkEncodedOrigin origin, int iw, int ih) {
+    SkScalar w = (SkScalar) iw;
+    SkScalar h = (SkScalar) ih;
     switch (origin) {
         case     kTopLeft_SkEncodedOrigin: return SkMatrix::I();
         case    kTopRight_SkEncodedOrigin: return SkMatrix::MakeAll(-1,  0, w,  0,  1, 0, 0, 0, 1);
