diff --git a/include/core/SkMatrix.h b/include/core/SkMatrix.h
index 96a12f111e..5ffd87e13d 100644
--- a/include/core/SkMatrix.h
+++ b/include/core/SkMatrix.h
@@ -93,7 +93,7 @@ public:
         return m;
     }
     [[nodiscard]] static SkMatrix Translate(SkVector t) { return Translate(t.x(), t.y()); }
-    [[nodiscard]] static SkMatrix Translate(SkIVector t) { return Translate(t.x(), t.y()); }
+    [[nodiscard]] static SkMatrix Translate(SkIVector t) { return Translate((SkScalar) t.x(), (SkScalar) t.y()); }
 
     /** Sets SkMatrix to rotate by |deg| about a pivot point at (0, 0).
 
