diff --git a/include/core/SkMatrix.h b/include/core/SkMatrix.h
index d36afa9af2..39df197f4a 100644
--- a/include/core/SkMatrix.h
+++ b/include/core/SkMatrix.h
@@ -85,7 +85,7 @@ public:
         return m;
     }
     [[nodiscard]] static SkMatrix Translate(SkVector t) { return Translate(t.x(), t.y()); }
-    [[nodiscard]] static SkMatrix Translate(SkIVector t) { return Translate(t.x(), t.y()); }
+    [[nodiscard]] static SkMatrix Translate(SkIVector t) { return Translate((SkScalar) t.x(), (SkScalar) t.y()); }
 
     [[nodiscard]] static SkMatrix ScaleTranslate(float sx, float sy, float tx, float ty);
 
