diff --git a/include/core/SkMatrix.h b/include/core/SkMatrix.h
index 03140760d6..ec7cd35923 100644
--- a/include/core/SkMatrix.h
+++ b/include/core/SkMatrix.h
@@ -86,7 +86,7 @@ public:
         return m;
     }
     static SkMatrix SK_WARN_UNUSED_RESULT Translate(SkVector t) { return Translate(t.x(), t.y()); }
-    static SkMatrix SK_WARN_UNUSED_RESULT Translate(SkIVector t) { return Translate(t.x(), t.y()); }
+    static SkMatrix SK_WARN_UNUSED_RESULT Translate(SkIVector t) { return Translate((SkScalar) t.x(), (SkScalar) t.y()); }
 
     /** Sets SkMatrix to rotate by |deg| about a pivot point at (0, 0).
 
